e2e/rescheduling/rescheduling.go

```diff
@@ -0,0 +1,388 @@
+package rescheduling
```

Comment:
Note to reviewers: these test cases directly port all the cases from the deleted `e2e/rescheduling/server_side_restarts_test.go`